pkg/cmd/workflow/view/view.go

```diff
@@ -0,0 +1,195 @@
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ViewOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	WorkflowSelector string
+	Web              bool
+
+	Prompt bool
+	Raw    bool
+}
+
+func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Command {
+	opts := &ViewOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+	cmd := &cobra.Command{
+		Use:    "view [<run-id>]",
+		Short:  "View a summary of a workflow run",
+		Args:   cobra.MaximumNArgs(1),
+		Hidden: true,
+		Example: heredoc.Doc(`
+		  # Interactively select a workflow to view
+		  $ gh workflow view
+
+		  # View a specific workflow
+		  $ gh workflow view 0451
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			opts.Raw = !opts.IO.CanPrompt()
+
+			if len(args) > 0 {
+				opts.WorkflowSelector = args[0]
+			} else if !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("workflow argument required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runView(opts)
+		},
+	}
+	cmd.Flags().BoolVarP(&opts.Web, "web", "w", false, "Open workflow in the browser")
+
+	return cmd
+}
+
+func runView(opts *ViewOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not build http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	var workflow *shared.Workflow
+
+	if opts.Prompt {
+		workflow, err = promptWorkflows(client, repo)
+		if err != nil {
+			return err
+		}
+	} else {
+		workflows, err := shared.ResolveWorkflow(client, repo, opts.WorkflowSelector)
+		if err != nil {
+			return err
+		}
+		if len(workflows) == 0 {
+			return fmt.Errorf("could not find any workflows named %s", opts.WorkflowSelector)
+		}
+
+		if len(workflows) == 1 {
+			workflow = &workflows[0]
+		} else {
+			if !opts.IO.CanPrompt() {
+				errMsg := "could not resolve to a unique workflow; found:"
+				for _, workflow := range workflows {
+					errMsg += fmt.Sprintf(" %s", workflow.Base())
+				}
+				return errors.New(errMsg)
+			}
+			states := []shared.WorkflowState{shared.Active}
+			workflow, err = shared.SelectWorkflow(workflows, "Which workflow do you mean?", states)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if opts.Web {
+		hostname := repo.RepoHost()
+		baseName := filepath.Base(workflow.Path)
+		workflowURL := fmt.Sprintf("https://%s/%s/actions/workflows/%s",
+			hostname,
+			ghrepo.FullName(repo),
+			baseName)
+		if opts.IO.IsStdoutTTY() {
+			fmt.Fprintf(opts.IO.Out, "Opening %s in your browser.\n", utils.DisplayURL(workflowURL))
+		}
+		return utils.OpenInBrowser(workflowURL)
```

Comment:
Just a heads up since this will likely conflict with https://github.com/cli/cli/pull/3279, who ever merges second will have to resolve the conflict.